### DISCUSSION TITLE
Um front-end precisa obrigatoriamente ser designer?

### DISCUSSION THREAD

**Discussion by claytonaalves** [frontendbr/forum#657] [Discussão]
A habilidade de design é um pré-requisito para se considerar front-end? Ou no mercado é comum a separação entre designer e front-end sendo este último apenas focado na "codificação" do layout (criado pelo designer) utilizando conhecimento de HTML, CSS e Javascript ?

**Comment by calaca** [comment]
Ao meu ver, o correto é separar design e front-end. O que vejo como necessário é ter algumas noções básicas de como utilizar certas ferramentas, como o Photoshop, já que vc pode precisar fazer alguma coisa lá durante um projeto. No mais, é isso mesmo que você mencionou.

**Comment by Wmitrut** [comment]
Olha, não necessariamente, mas ter noções de design é algo que ajuda e muito. Eu fico nesse meio termo mesmo, adoro projetar UIs no Figma e Adobe XD, estudar sobre noções básicas de usabilidade, sou uma pessoa sarna com uma boa UI. **Mas pessoalmente não é o suficiente só o design**,  pra mim, eu gosto de fazer funcionar, ver o que projetei realmente ganhando vida.

**Comment by kvnol** [comment]
Conforme a @calaca disse:

> O que vejo como necessário é ter algumas noções básicas

É importante o front-end ter noções dos quatro princípios básicos do design que são: **proximidade**, **repetição**, **alinhamento** e **contraste** - recomendo a leitura do artigo [Os 4 princípios básicos de design](http://www.midializado.com.br/2011/07/os-4-principios-basicos-de-design.html) e do livro [Design para quem não é designer](https://www.amazon.com.br/Design-Para-Quem-n%C3%A3o-Designer/dp/857416836X) - para que, na aplicação na tela que o designer te passa você não fique perdido com alinhamentos e até em casos que você pega o layout num .psd único, sem versões mobile, você não se perca e consiga ajustar o layout da melhor forma.

Agora é certo que habilidades de UI Design e UX para front-end é um grande diferencial na hora de conseguir uma vaga ou crescimento dentro da empresa no caso de você, como front-end, conseguir conversar de igual para igual com os designers e até dar dicas sobre o que foi pensado por eles.

**Comment by calaca** [comment]
Apenas completando as dicas do @kvnol: também seria legal dar uma olhada nas boas e velhas [10 heurísticas de Nielsen](http://blog.caelum.com.br/10-heuristicas-de-nielsen-uma-formula-pra-evitar-erros-basicos-de-usabilidade/) e nas técnicas de [gestalt](https://www.chiefofdesign.com.br/gestalt/). Ambas vão te dar uma noção básica bem legal de design.

**Comment by Wmitrut** [comment]
>  seria legal dar uma olhada nas boas e velhas 10 heurísticas de Nielsen e nas técnicas de gestalt. 

Dica certeira @calaca, outra dica que dou é olhar  as documentações de sistemas de design como o Material, o do AirBnB, enfim... guias de estilo, elas são essenciais pra quem ta querendo ter boas noções e achar desafios de implementação

**Comment by kazzkiq** [comment]
Na teoria, não.

Na prática, é importante ter pelo menos noções mínimas de design, e proficiência nos programas usados pra criar layouts, já que nem sempre o designer do time vai estar disponível pra exportar ícones e recortar imagens do layout pra você (isso quando o time *tem* um designer exclusivo), e você não quer ser o culpado de atrasar a entrega do projeto.

**Comment by felipefialho** [comment]
Acho que só noção estética para que os projetos fiquem similares ao layout original (de preferencia até melhor!). 

Acho que boas noções de UX é mais importante do que de Design.

**Comment by lucianobarauna** [comment]
O design agrega bastante o front-end ainda mais como @Wmitrut comentou ali em cima. Se por acaso um designer não souber fechar um arquivo svg por exemplo você com conhecimento de design saberia como resolver, 
Hoje sou front-end, formado em design e fiz a faculdade para complementar meu front mais por conta de algumas vagas vejo que vou ter que fazer uma de TI rs ;-)

**Comment by maclovin** [comment]
**Resposta curta:** designer by the book, acredito que sim.
**Resposta longa:** depois de um certo tempo eu comecei a identificar 2 tipos de profissionais na nossa área, o que se limita ao combo (html, css, js) e o que entende que front-end é algo muito maior do que simplesmente entregar projetos pre-definidos, aquele que pega um problema e o resolve de uma forma inovadora ou disruptiva. Onde que eu quero chegar é, o desenvolvedor frontend tem um papel muito maior do que simplesmente implementar um layout performático. Eu acredito que o nosso papel no geral é diminuir ao máximo a distancia/complexidade entre a tecnologia da informação e o ser-humano, pensando na melhor forma de fazer o I/O dentro de um determinado contexto. Concluindo, eu acho que um estudo sobre a história das interfaces gráficas, estudo sobre processos de design (de produto, de material, de layout), como utilizar o máximo do hardware, e comportamento do ser humano possam abrir a nossa cabeça para melhor compreendermos como resolver determinados tipos de problemas.

**Comment by Guepardo** [comment]
Na minha opinião, não existe obrigatoriedade em ser designer pra ser front-end. Da mesma que um programador não precisa saber arquitetura de software, uma framework já basta pra muitos. 

Mas quando o adjetivo **ser um bom** é adicionado a pegunta, a relação entre ser front-end e designer é quase paradoxal: 

> Para ser um bom front-end é necessário ter boas noções de designer e para ser um bom designer para web é necessário ter boas noções de front-end.

Se limitar apenas na codificação de um layout te torna um profissional mediano e substituível.
